File: 2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        fre, res = [], []
        cnt = 0
        vowels = "aeiou"
        for word in words:
            if word[0] in vowels and word[len(word) - 1] in vowels:
                cnt += 1
            fre.append(cnt)
                
        for l, r in queries:
            if l == 0:
                res.append(fre[r])
            else:
                res.append(fre[r] - fre[l - 1])
    
        return res     

File: 2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        vector<int> fre, res;
        int cnt = 0;
        string vowels = "aeiou";
        
        for (const auto& word : words) {
            if (vowels.find(word[0]) != string::npos && vowels.find(word[word.length() - 1]) != string::npos) {
                cnt += 1;
            }
            fre.push_back(cnt);
        }
        
        for (const auto& query : queries) {
            int l = query[0], r = query[1];
            if (l == 0) {
                res.push_back(fre[r]);
            } else {
                res.push_back(fre[r] - fre[l - 1]);
            }
        }
        return res;
    }
};